src/test/fetch.request.test.ts

```diff
@@ -48,13 +60,6 @@ describe('fetch request', () => {
       await assert200(fetch(url('/', { foo: '123', bar: '456' })));
     });
 
-    it('merges query string parameters', async () => {
-      expect.assertions(1);
-      interceptor.intercept('/', 'get').query({ foo: '123', bar: '456' }).reply(200);
-
-      const fetch = createFetch(got.extend({ searchParams: { bar: '456' } }));
-      await assert200(fetch(url('/', { foo: '123' })));
-    });
```

Comment:
I think disabling this might be problematic because I'd expect people to use it to add access tokens/API keys as a base on their clients :thinking: